src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -164,11 +218,12 @@ public ReadsDataSource( final List<Path> samPaths, final List<Path> samIndices,
 
             SamReader reader;
             if ( samIndices == null ) {
-                reader = samReaderFactory.open(samPath);
+                reader = samReaderFactory.open(samPath, wrapper, indexWrapper);
```

Comment:
Is there also a way we can add a test to check that when indices are not explicitly specified and the bam is on GCS, index wrappers get applied?